pkg/provisioner/ironic/ironic.go

```diff
@@ -1336,6 +1340,34 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		p.log.Info("host is being powered off before removal")
+		provResult, err := p.hardPowerOff()
+		if err != nil {
+			switch err.(type) {
+			case HostLockedError:
+				result.RequeueAfter = powerRequeueDelay
+				result.Dirty = true
+				return result, err
+			default:
+				if result.ErrorCount < maxErrorRetryCount {
+					result.ErrorCount++
+					result.Dirty = true
+					return result, err
+				}
+				result.Dirty = false
+			}
+
+		} else {
+			// No errors returned by hardPowerOff
+			if provResult.Dirty {
+				result.RequeueAfter = powerRequeueDelay
+				result.Dirty = true
+				return result, err
+			}
+			result.Dirty = false
```

Comment:
I think this is probably not needed, either.

Reply:
Done.